lib/text/vtt_text_parser.js

```diff
@@ -221,96 +221,112 @@ shaka.text.VttTextParser = class {
-    if (text[1].includes('}')) {
-      const payload = /\{(.*?)\}/.exec(text[1]);
-      if (payload) {
-        propertyLines = payload[1].split(';');
+    for (const styleBlock of styleBlocks) {
+      let styleSelector = 'global';
+      // Look for what is within parentisesis. For example:
+      // <code>:: cue (b) {</code>, what we are looking for is <code>b</code>
+      const selector = styleBlock[0].match(/\((.*)\)/);
+      if (selector) {
+        styleSelector = selector.pop();
       }
-    }
 
-    const cue = new shaka.text.Cue(0, 0, '');
-    let validStyle = false;
-    for (let i = 0; i < propertyLines.length; i++) {
-      // We look for CSS properties. As a general rule they are separated by
-      // <code>:</code>. Eg: <code>color: red;</code>
-      const lineParts = /^\s*([^:]+):\s*(.*)/.exec(propertyLines[i]);
-      if (lineParts) {
-        const name = lineParts[1].trim();
-        const value = lineParts[2].trim().replace(';', '');
-        switch (name) {
-          case 'background-color':
-            validStyle = true;
-            cue.backgroundColor = value;
-            break;
-          case 'color':
-            validStyle = true;
-            cue.color = value;
-            break;
-          case 'font-family':
-            validStyle = true;
-            cue.fontFamily = value;
-            break;
-          case 'font-size':
-            validStyle = true;
-            cue.fontSize = value;
-            break;
-          case 'font-weight':
-            if (parseInt(value, 10) >= 700) {
+      // We start at 1 to avoid '::cue' and end earlier to avoid '}'
+      let propertyLines = styleBlock.slice(1, -1);
+      if (styleBlock[0].includes('}')) {
+        const payload = /\{(.*?)\}/.exec(styleBlock[0]);
+        if (payload) {
+          propertyLines = payload[1].split(';');
+        }
+      }
+
+      const cue = new shaka.text.Cue(0, 0, '');
+      let validStyle = false;
+      for (let i = 0; i < propertyLines.length; i++) {
+        // We look for CSS properties. As a general rule they are separated by
+        // <code>:</code>. Eg: <code>color: red;</code>
+        const lineParts = /^\s*([^:]+):\s*(.*)/.exec(propertyLines[i]);
+        if (lineParts) {
+          const name = lineParts[1].trim();
+          const value = lineParts[2].trim().replace(';', '');
+          switch (name) {
+            case 'background-color':
               validStyle = true;
-              cue.fontWeight = shaka.text.Cue.fontWeight.BOLD;
-            }
-            break;
-          case 'font-style':
-            switch (value) {
-              case 'normal':
-                validStyle = true;
-                cue.fontStyle = shaka.text.Cue.fontStyle.NORMAL;
-                break;
-              case 'italic':
-                validStyle = true;
-                cue.fontStyle = shaka.text.Cue.fontStyle.ITALIC;
-                break;
-              case 'oblique':
+              cue.backgroundColor = value;
```

Comment:
This test case uses "background" instead of "background-color".  Could you support that, too, even just making 'background' an alias to 'background-color' for now?

```webvtt
WEBVTT

STYLE
*|*::cue(b) {
  background: blue
}
|*::cue(i) {
  color: blue;
}
::cue(i) {
  background: white;
}
*::cue(b) {
  color: white;
}
::cue {
  font-size: 11px;
}

00:00:00.000 --> 00:00:05.000
This <i>is</i> a <b>test</b> subtitle
```

Reply:
Added!